src/azul/drs.py

```diff
@@ -74,10 +87,158 @@ class Access:
     headers: Optional[Mapping[str, str]] = None
 
 
+class DRSURI(metaclass=ABCMeta):
+
+    @property
+    @abstractmethod
+    def _object_id(self) -> str:
+        """
+        The portion of the URI that identifies the data object.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def parse(cls, drs_uri: str) -> 'DRSURI':
+        try:
+            drs_uri = RegularDRSURI.parse(drs_uri)
+        except ValueError as e:
+            if len(e.args) == 1 and isinstance(e.args[0], str) and e.args[0].startswith('Invalid port '):
+                drs_uri = CompactDRSURI.parse(drs_uri)
+            else:
+                raise
```

Comment:
This tightly coupled with the use of `furl` in RegularDRSURI.parse and therefore brittle but also quite obscure.

I would call the `furl` constructor here, invoke the  RegularDRSURI constructor with it and eliminate RegularDRSURI.parse altogether.

Reply:
Decided at PL to abandon the try/except strategy for detecting compact identifiers. The coupling has been eliminated.

I decided not to remove `RegularDRSURI.parse` because if it inherits the superclass implementation, it could counter-intuitively return instances of `CompactDRSURI`.